packages/zod/src/v4/core/tests/locales/he.test.ts

```diff
@@ -0,0 +1,370 @@
+      expect(result.success).toBe(false);
+      if (!result.success) {
+        expect(result.error.issues[0].message).toBe("קלט לא תקין: הערך בוליאני צריך להיות, התקבל ערך ריק (null)");
+      }
+    });
+
+    test("array expected (masculine), object received", () => {
+      const schema = z.array(z.string());
+      const result = schema.safeParse({});
+      expect(result.success).toBe(false);
+      if (!result.success) {
+        expect(result.error.issues[0].message).toBe("קלט לא תקין: המערך צריך להיות, התקבל אובייקט");
+      }
+    });
+
+    test("object expected (masculine), array received", () => {
+      const schema = z.object({ a: z.string() });
+      const result = schema.safeParse([]);
+      expect(result.success).toBe(false);
+      if (!result.success) {
+        expect(result.error.issues[0].message).toBe("קלט לא תקין: האובייקט צריך להיות, התקבל מערך");
+      }
+    });
+
+    test("function expected (feminine), string received", () => {
+      const schema = z.function();
+      const result = schema.safeParse("not a function");
+      expect(result.success).toBe(false);
+      if (!result.success) {
+        expect(result.error.issues[0].message).toBe("קלט לא תקין: הפונקציה צריכה להיות, התקבל מחרוזת");
+      }
+    });
+  });
+
+  describe("gendered verbs consistency", () => {
+    test("feminine types use צריכה", () => {
+      const feminineTypes = [
+        { schema: z.string().min(5), input: "abc" },
+        { schema: z.set(z.string()).min(2), input: new Set(["a"]) },
+      ];
+
+      for (const { schema, input } of feminineTypes) {
+        const result = schema.safeParse(input);
+        expect(result.success).toBe(false);
+        if (!result.success) {
+          expect(result.error.issues[0].message).toContain("צריכה");
+        }
+      }
+    });
+
+    test("masculine types use צריך", () => {
+      const masculineTypes = [
+        { schema: z.number().min(10), input: 5 },
+        { schema: z.array(z.string()).min(2), input: ["a"] },
+      ];
+
+      for (const { schema, input } of masculineTypes) {
+        const result = schema.safeParse(input);
+        expect(result.success).toBe(false);
+        if (!result.success) {
+          expect(result.error.issues[0].message).toContain("צריך");
+        }
+      }
+    });
+  });
+
+  describe("invalid_value with enum", () => {
+    test("single value", () => {
+      const schema = z.enum(["a"]);
+      const result = schema.safeParse("b");
+      expect(result.success).toBe(false);
+      if (!result.success) {
+        expect(result.error.issues[0].message).toBe('קלט לא תקין: צריך "a"');
+      }
+    });
+
+    test("multiple values", () => {
+      const schema = z.enum(["a", "b", "c"]);
+      const result = schema.safeParse("d");
+      expect(result.success).toBe(false);
+      if (!result.success) {
+        expect(result.error.issues[0].message).toBe('קלט לא תקין: צריך אחת מהאפשרויות "a" | "b" | "c"');
```

Comment:
```suggestion
        expect(result.error.issues[0].message).toBe('קלט לא תקין: האפשרויות המתאימות הן "a", "b", או "c"');
```

wdyt?
others as well
(maybe still need to think about how it'll look like in singular form)